ruma-api-macros/src/api/metadata.rs

```diff
@@ -54,6 +63,10 @@ fn set_field<T: ToTokens>(field: &mut Option<T>, value: T) -> syn::Result<()> {
     }
 }
 
+fn add_field<T: ToTokens>(field: &mut Vec<MetadataField<T>>, value: T, attrs: Vec<Attribute>) {
```

Comment:
I don't think this needs to be a separate function, you can just inline it at the two usage sites.